enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/APIFactory.java

```diff
@@ -91,8 +101,32 @@ public void removeApi(API api) {
         apis.remove(apiKey);
     }
 
+    /**
+     * Validate the GraphQL schema before adding the API.
+     *
+     * @param api api details
+     * @return whether the api is an valid graphQL API.
+     */
+    private boolean isValidGraphQLAPI(Api api) {
+        if (!api.getGraphQLSchema().isEmpty()) {
+            String sdl = api.getGraphQLSchema();
+            SchemaParser schemaParser = new SchemaParser();
+            try {
+                schemaParser.parse(sdl);
+                return true;
+            } catch (SchemaProblem exception) {
+                logger.error("Error while parsing graphQL sdl for API : {} version : {}", api.getTitle(),
+                        api.getVersion(),
+                        ErrorDetails.errorLog(LoggingConstants.Severity.MAJOR, 5201), exception);
+            }
+        } else {
+            logger.error("GraphQL sdl for API : {} version : {} is empty", api.getTitle(), api.getVersion(),
```

Comment:
shall we print apiUUID too?

Reply:
We can remove this block too now as adapter validate if the sdl is empty. @slahirucd7 Am I right?